libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -430,12 +431,35 @@ AP_GPS::update_instance(uint8_t instance)
     } else {
         timing[instance].last_message_time_ms = tnow;
         if (state[instance].status >= GPS_OK_FIX_2D) {
-            timing[instance].last_fix_time_ms = tnow;
+            estimate_fix_time(instance, tnow);
         }
     }
 }
 
 /*
+  estimate the fix time on the sensor in terms of the system clock in
+  milliseconds
+ */
+void AP_GPS::estimate_fix_time(uint8_t instance, uint32_t now)
+{
+    const uint64_t ms_per_week = 604800000ULL;
```

Comment:
In general I'd rather see 7 * 24 * 60 * 60 * 1000 for magic constants like this as it makes it a bit more apparent where the number came from in general (this one is fairly self explanatory and is obviously correct)